Dockerfile

```diff
@@ -26,6 +27,10 @@ RUN --mount=type=cache,target="/root/.cache/go-build" \
     go build -tags "${GO_BUILD_TAGS}" -trimpath -ldflags="${GO_LINKER_ARGS}" \
     -o ./build/_output/bin/dynatrace-operator ./cmd/
 
+# Generate binary SBOM - improved hierarchy resolution
+RUN go install github.com/CycloneDX/cyclonedx-gomod/cmd/cyclonedx-gomod@v1.9.0
+RUN cyclonedx-gomod app -licenses -assert-licenses -json -main ./cmd -output ./build/_output/bin/dynatrace-operator-bin-sbom.json
```

Comment:
did some testing, and IMO this shouldn't be done here, we don't want this to be part of the build every time for every image, but only for release images.

I timed our `make build`, which is the local build for development, and it doesn't look good:
```
//with SBOM gen
make build  12.24s user 6.39s system 11% cpu 2:49.15 total
make build  12.12s user 6.22s system 11% cpu 2:40.21 total

//without SBOM gen
make build  10.31s user 4.92s system 17% cpu 1:26.53 total
make build  10.11s user 4.80s system 18% cpu 1:18.91 total
```
- So basically doubled the build time, which I reeeeally want to avoid

Reply:
👍 good point